Project/core/Payment.cs

```diff
@@ -6,64 +6,41 @@
 
 namespace Project
 {
+    [Serializable]
     class Payment
     {
-        private Client client;
-        private List<SinglePayment> singlePayments;
+        private DateTime date;
+        private string title;
+        private bool isPaid;
+        private double amount;
 
         //Konstruktory 
 
-        public Payment(Client client)
+        public Payment(string title)
         {
-            this.client = client;
-            this.singlePayments = new List<SinglePayment>();
+            this.date = DateTime.Now;
+            this.title = title;
+            this.isPaid = false;
+            this.amount = Amount();
```

Comment:
W tym momencie to zostanie wywołane przy tworzeniu płatności, i nie będzie się updatować, po to jest metoda abstrakcyjna Amount żeby z niej korzystać.
Ta zmienna prywatna amount jest nie potrzebna.